src/plugins/mer/merbuildconfiguration.cpp

```diff
@@ -56,6 +56,9 @@ namespace {
 
 const char MER_VARIABLES_CACHE_FILENAME[] = ".mb2/qmake_variables.cache";
 
+// Avoid actions on temporary changes and asking questions too hastily
+const int UPDATE_EXTRA_PARSER_ARGUMENTS_DELAY_MS = 3000;
```

Comment:
I think there are some cases (e.g. when opening a project) when we should not wait. It's rather annoying when the dialog interrupts you when you had already started e.g. opening another project or something.

Reply:
Unfortunately there are  too many scenarios to consider. Opening a project is actualy one of those that need some sort of care - be it the simple delay or something more complicated. When a project is being opened there are temporary states when it is unconfigured or configured using the default kit or so.

Ideally we should avoid the necessity for asking anything here, but I don't have a better idea how to reliably collect the additional args now. We should think about this more a propose some future work.

Reply:
I agree. I'll write it down on my own notes so that I can bring it up again in the future :)